### PR TITLE
Correct clipping

### PR DESCRIPTION
**Pull request by Kalaboum**

### CODE REVIEW

File: src/modules/dataImport/clipping.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+function coordsInBbox(coords, merc, x, y, z) {
+  const bbox = merc.bbox(x, y, z);
+  const line = turf.lineString(coords);
+  const bboxLineString = bboxToLineString(bbox);
+  return turf.lineIntersect(line, bboxLineString).features.length > 0;
+}
+
+function unitTest() {
```
Comment (romanzoller) [inline src/modules/dataImport/clipping.js]: Can this be moved to a "real" unit test in another file?

File: src/modules/dataImport/cuttingShapes.js [hunk 2]
```diff
@@ -267,10 +267,12 @@ function removeDuplicatesInSortedShape(shapePoints) {
...
 * stoptimes with shape_dist_traveled
 */
-// returns a fractionShape with empty dictionary if stopTimesList is emptym can be a problem
+// returns a fractionShape with empty dictionary if stopTimesList is empty can be a problem
 function fractionShape(shapePoints, stopTimesList) {
   sortShape(shapePoints);
-  const shapePointsFiltered = removeDuplicatesInSortedShape(shapePoints);
+  const shapePointsFiltered = shapePoints; //removeDuplicatesInSortedShape(shapePoints);
+  // removeDuplicates can cause bugs, it may removes the corresponding shapeDistTraveled 
+  // leading to a false fractioning of the shape
```
Comment (romanzoller) [inline src/modules/dataImport/cuttingShapes.js]: Didn't the duplicate cause some kind of problem?

File: src/modules/dataImport/exportSchedule.js [hunk 3]
```diff
@@ -111,7 +111,8 @@ async function exportScheduleToGeoJSON() {
   await connect(config.mongoUrl);
   const servicesIds = await getServicesIds(config);
   const agencyKeys = config.agencies.map(agency => agency.agency_key);
-  let shapesIds = await gtfs.getShapeIds();
+  // let shapesIds = await gtfs.getShapeIds();
+  let shapesIds = ['721761'];
```
Comment (romanzoller) [inline src/modules/dataImport/exportSchedule.js]: This looks like a leftover of some test hack...

File: src/modules/dataImport/cuttingShapes.js (hunk 2 above)
Comment (Kalaboum) [reply src/modules/dataImport/cuttingShapes.js]: Duplicates points in shapes is not an issue, it was a misconception from me

**Pull request merged**